install-k6.js

```diff
@@ -0,0 +1,91 @@
+const { exec } = require('child_process')
+const { existsSync } = require('fs')
+
+const K6_VERSION = 'v0.55.0'
+const K6_PATH_MAC_AMD = `k6-${K6_VERSION}-macos-amd64`
+const K6_PATH_MAC_ARM = `k6-${K6_VERSION}-macos-arm64`
+const K6_PATH_WIN_AMD = `k6-${K6_VERSION}-windows-amd64`
+
+const existsK6 = (os, arch) => {
+  return existsSync(`resources/${os}/${arch}/k6`)
+}
+
+const getMacOSK6Binary = () => {
+  const command = `
+# download binaries
+curl -LO https://github.com/grafana/k6/releases/download/${K6_VERSION}/${K6_PATH_MAC_AMD}.zip
+curl -LO https://github.com/grafana/k6/releases/download/${K6_VERSION}/${K6_PATH_MAC_ARM}.zip
+
+# unzip & smoke test
+unzip ${K6_PATH_MAC_AMD}.zip
+unzip ${K6_PATH_MAC_ARM}.zip
+${K6_PATH_MAC_AMD}/k6 version
+${K6_PATH_MAC_ARM}/k6 version
+
+# move to resource folder
+mv ${K6_PATH_MAC_AMD}/k6 resources/mac/x86_64
+mv ${K6_PATH_MAC_ARM}/k6 resources/mac/arm64
+
+# cleanup
+rm ${K6_PATH_MAC_AMD}.zip
+rm ${K6_PATH_MAC_ARM}.zip
+rmdir ${K6_PATH_MAC_AMD}
+rmdir ${K6_PATH_MAC_ARM}
+`
+
+  executeCommand(command, {})
+}
+
+const getWindowsK6Binary = () => {
+  const command = `
+# download binaries
+Invoke-WebRequest -Uri "https://github.com/grafana/k6/releases/download/${K6_VERSION}/${K6_PATH_WIN_AMD}.zip" -OutFile "k6-windows-amd64.zip"
+
+# unzip & smoke test
+Expand-Archive -Path "k6-windows-amd64.zip" -DestinationPath "."
+${K6_PATH_WIN_AMD}\\k6.exe version
+
+# move to resource folder
+Move-Item -Path "${K6_PATH_WIN_AMD}\\k6.exe" -Destination resources\\win\\x86_64
+`
+
+  executeCommand(command, { shell: 'powershell.exe' })
+}
+
+const executeCommand = (command, options) => {
+  exec(command, options, (error, stdout, stderr) => {
+    if (error) {
+      console.error(`Error: ${error.message}`)
+      return
+    }
+
+    if (stderr) {
+      console.error(`stderr: ${stderr}`)
+      return
+    }
+
+    console.log(`stdout: ${stdout}`)
+  })
+}
+
+switch (process.platform) {
+  case 'darwin':
+    // we check only for one arch since we include both binaries
+    if (!existsK6('mac', 'arm64')) {
+      console.log('k6 binary not found')
+      console.log('downloading k6... this might take some time...')
+      getMacOSK6Binary()
+      console.log('k6 binary download completed')
+    }
+    break
+  case 'win32':
+    if (!existsK6('win', 'x86_64')) {
+      console.log('k6 binary not found')
+      console.log('downloading k6... this might take some time...')
+      getWindowsK6Binary()
+      console.log('k6 binary download completed')
```

Comment:
I noticed a race condition between downloading the binaries and showing the success message while it was still in progress. We can solve this by making `executeCommand` resolve or reject a Promise or, we can move the success message inside the `exec` callback.

![image](https://github.com/user-attachments/assets/2983b5b8-f00b-4a8e-b850-771c8bb8223c)

Reply:
Noticed this one as well!
I'm now making use of `execSync` instead of `exec` and that fixes the message issue as well 🙌